cmd/oras/root/cp.go

```diff
@@ -269,7 +269,33 @@ func prepareCopyOption(ctx context.Context, src oras.ReadOnlyGraphTarget, dst or
 
 	if len(referrers) == 0 {
 		// no child referrers
-		return opts, nil
+		return opts, root, nil
+	}
+
+	rootReferrers, err := opts.FindPredecessors(ctx, src, root)
```

Comment:
Since we have already found predecessors on root, we can save one call at line 303 for root.